scripts/unreleased-changes

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+"""
+Shell script to help find unreleased changes on release branches.
+
+Loops through each release branch from current branch to 1.0 and prints out
+the changes that have not been released yet in json format as:
+
+{"branch": "1.1","changes": ["..."]}
+{"branch": "1.0","changes": ["..."]}
+"""
+import json
+import packaging.version
+import subprocess
+import sys
+
+
+def get_release_branches() -> list[str]:
+    # get list of all git release tags
+    tags = subprocess.check_output(["git", "tag", "--list", "v*"]).splitlines()
+    # get list of all release branches
+    branches = sorted(
+        {".".join(tag.decode("utf-8")[1:].split(".")[:2]) for tag in tags},
+        key=lambda x: packaging.version.Version(x),
+        reverse=True,
+    )
+
+    return [branch for branch in branches if float(branch) >= 1.0]
+
+
+def get_unreleased_changes(branch: str) -> list[str]:
+    # get list of reno changes for the branch
+    results = subprocess.check_output(
+        ["reno", "--quiet", "list", "--branch", branch, "--stop-at-branch-base"], stderr=subprocess.PIPE
+    ).decode("utf-8")
+
+    changes = []
+
+    unreleased_version = None
+    for line in results.splitlines():
+        if line.startswith("v") and "-" in line:
+            unreleased_version = line.strip()
+        elif line.startswith("\t") and unreleased_version:
+            changes.append(line.strip())
+        elif unreleased_version and line.startswith("v"):
```

Comment:
NB: What case is this covering?

Reply:
added a code comment above, maybe that helps?